src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs

```diff
@@ -245,9 +248,10 @@ protected async Task<Diagnostic[]> GetSortedDiagnosticsFromDocumentsAsync(Analyz
 			var allDiagnostics = await compilationWithAnalyzers.GetAllDiagnosticsAsync();
 			var errors = allDiagnostics.Where(d => d.Severity == DiagnosticSeverity.Error).ToList();
 			foreach (var error in errors)
-			{
 				Assert.True(false, $"Line {error.Location.GetLineSpan().StartLinePosition.Line}: {error.GetMessage()}");
-			}
+
+			foreach (var analyzerException in analyzerExceptions)
+				Assert.True(false, analyzerException.Message);
```

Comment:
This reads a bit strange. `Assert.Fail` maybe? We'll also see only the first analyzer exception. Maybe assert that the count of the analyzer Exception is zero, and if that fails the message if the aggregate representation of all exceptions.

Reply:
https://github.com/xunit/xunit/issues/2027